lib/graphql/internal_representation/rewrite.rb

```diff
@@ -137,9 +141,9 @@ def validate(context)
         # can be shared between its usages.
         context.on_dependency_resolve do |defn_ast_node, spread_ast_nodes, frag_ast_node|
           frag_name = frag_ast_node.name
-          fragment_node = fragment_definitions[frag_name]
+          fragment_node = @definitions[frag_name]
 
-          if fragment_node
+          if fragment_node && fragment_node.ast_node.is_a?(Nodes::FragmentDefinition)
```

Comment:
Does something break without this check? (I wonder what _happens_ if we have operations and fragments with the same name ... :S )